worlds/banjo_tooie/__init__.py

```diff
@@ -0,0 +1,1094 @@
+            regionName.TLBOSS: regionName.TLTOP,
+            regionName.GIBOSS: regionName.GI1,
+            regionName.HPFBOSS: regionName.HP,
+            regionName.HPIBOSS: regionName.HP,
+            regionName.CCBOSS: regionName.CC,
+        }
+        bt_players = world.get_game_players(cls.game)
+        spoiler_handle.write('\n\nBanjo-Tooie ({})'.format(BanjoTooieWorld.version))
+        for player in bt_players:
+            currentWorld: BanjoTooieWorld = world.worlds[player]
+            name = world.get_player_name(player)
+            spoiler_handle.write(f"\n\n{name}:")
+            spoiler_handle.write('\n\tLoading Zones:')
+            for starting_zone, actual_world in currentWorld.loading_zones.items():
+                if actual_world == regionName.JR:
+                    spoiler_handle.write(f"\n\t\t{entrance_hags[starting_zone]} -> Jolly Roger's Lagoon")
+                elif actual_world == regionName.GIO:
+                    spoiler_handle.write(f"\n\t\t{entrance_hags[starting_zone]} -> Grunty Industries")
+                else:
+                    spoiler_handle.write(f"\n\t\t{entrance_hags[starting_zone]} -> {actual_world}")
+            spoiler_handle.write("\n\tWorld Requirements:")
+            for entrances, cost in currentWorld.world_requirements.items():
+                if entrances == regionName.JR:
+                    spoiler_handle.write(f"\n\t\tJolly Roger's Lagoon: {cost}")
+                elif entrances == regionName.GIO:
+                    spoiler_handle.write(f"\n\t\tGrunty Industries: {cost}")
+                else:
+                    spoiler_handle.write(f"\n\t\t{entrances}: {cost}")
+            spoiler_handle.write("\n\tJamjars' Silo Costs:")
+            for silo, cost in currentWorld.jamjars_siloname_costs.items():
+                spoiler_handle.write(f"\n\t\t{silo}: {cost}")
+            spoiler_handle.write('\n\tHints:')
+            for location_id, hint_data in currentWorld.hints.items():
+                spoiler_handle.write("\n\t\t{}: {}".format(
+                    currentWorld.location_id_to_name[location_id],
+                    hint_data.text
+                ))
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        t0 = time.time()
+        generate_hints(self)
+        t1 = time.time()
+        total = t1-t0
+        if total >= 1:
+            logging.info(f"Took {total:.4f} seconds in BanjoTooieWorld.generate_hints for player {self.player}, named {self.multiworld.player_name[self.player]}.")
+        btoptions = {option_name: option.value for option_name, option in self.options.__dict__.items()}
+
+        # TODO: AP 0.6.3: plando not serialisable, so we can't include it in slot_data. Remove this line when 0.6.4 goes live.
+        btoptions.pop("plando_items")
+
+        # Elements that are randomised outside the yaml and affects gameplay
+        custom_bt_data: Dict[str, Any] = {
+            "player_name": self.multiworld.player_name[self.player],
```

Comment:
```suggestion
            "player_name": self.player_name,
```
There's a helper for this now.